files/tests/test_race_conditions.py

```
@@ -0,0 +1,418 @@
+    
+    def test_rate_limiting_blocks_spam(self):
+        """
+        Test: Rate limiting blocks spam after 30 rapid views within 5 seconds.
+        Expected: First 30 views allowed, 31st blocked and NOT recorded.
+        """
+        media = self.test_media
+        spam_ip = '203.0.113.99'
+        max_views = getattr(settings, 'MAX_ANONYMOUS_VIEWS_PER_5SEC', 30)
+        
+        print(f"\n🚫 Testing rate limiting: {max_views} views/5sec from IP {spam_ip}")
+        print(f"   (Allows classrooms of 30+ students while blocking spam)")
```

Ruff (0.14.0)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)